cpp/Platform.Converters/CachingConverterDecorator.h

```diff
@@ -1,15 +1,28 @@
-﻿namespace Platform::Converters
+﻿#pragma once
+#include <unordered_map>
+#include <memory>
+
+#include "IConverter[TSource, TTarget].h"
+
+namespace Platform::Converters
 {
     template <typename ...> class CachingConverterDecorator;
     template <typename TSource, typename TTarget> class CachingConverterDecorator<TSource, TTarget> : public IConverter<TSource, TTarget>
     {
-        private: readonly IConverter<TSource, TTarget> *_baseConverter;
-        private: readonly IDictionary<TSource, TTarget> *_cache;
-
-        public: CachingConverterDecorator(IConverter<TSource, TTarget> &baseConverter, IDictionary<TSource, TTarget> &cache) { (_baseConverter, _cache) = (baseConverter, cache); }
+        private: std::unique_ptr<IConverter<TSource, TTarget>> _baseConverter;
+        private: std::unordered_map<TSource, TTarget> _cache;
 
-        public: CachingConverterDecorator(IConverter<TSource, TTarget> &baseConverter) : this(baseConverter, Dictionary<TSource, TTarget>()) { }
+        public: CachingConverterDecorator(IConverter<TSource, TTarget>& baseConverter, std::unordered_map<TSource, TTarget>& cache): _baseConverter(baseConverter), _cache(cache) {}
+        public: CachingConverterDecorator(IConverter<TSource, TTarget> &baseConverter): _baseConverter(baseConverter), _cache() {}
```

Comment:
There should be a single empty line separator between all methods, including constructors.